sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GoneAndRetryWithRetryPolicy.java

```diff
@@ -154,6 +154,8 @@ public GoneAndRetryWithRetryPolicy(RxDocumentServiceRequest request, Integer wai
                         .error(BridgeInternal.createCosmosClientException(HttpConstants.StatusCodes.SERVICE_UNAVAILABLE, exception)));
             }
 
+            // Note: we access request above without any null check then another null check here doesn't make sense.
+            // @kushagraThapar/@moderakh/@mbhaskar
             if (this.request != null) {
```

Comment:
True that, why don't we get this null check out of the way ?

Reply:
this.request is not accessed in all if/else branches. is it?

can we discuss this outside of scope of this PR?

Reply:
sounds good, reverted.